packages/block-library/src/navigation/index.php

```diff
@@ -257,10 +257,16 @@ function block_core_navigation_get_classic_menu_fallback() {
 	$classic_nav_menus = wp_get_nav_menus();
 
 	// If menus exist.
-	if ( $classic_nav_menus && ! is_wp_error( $classic_nav_menus ) && count( $classic_nav_menus ) === 1 ) {
-		// Use the first classic menu only. Handles simple use case where user has a single
-		// classic menu and switches to a block theme. In future this maybe expanded to
-		// determine the most appropriate classic menu to be used based on location.
+	if ( $classic_nav_menus && ! is_wp_error( $classic_nav_menus ) ) {
+		// Handles simple use case where user has a classic menu and switches to a block theme.
+
+		// Attempts to return the menu assigned to location `primary` (a convention many classic themes follow).
+		$locations = get_nav_menu_locations();
+		if ( isset( $locations[ 'primary' ] ) ) {
```

Comment:
Do we need to check whether $locations is not empty?

Reply:
The function returns an empty array if they are false.

Calling `isset` on a key in empty array should result in `false`.

I suppose a simple test would be to remove all your classic menus and run the code.

Reply:
> Calling isset on a key in empty array [should result in false](https://onlinegdb.com/8PNUVTo9PS).

👍 

> I suppose a simple test would be to remove all your classic menus and run the code.

The test I ran to verify was switching to a block theme the following conditions:
- the block theme has no menu locations defined
- no classic menus defined
- no published navigation menus
- viewing the front-end and verifying no errors